Errors are thrown, when Pestermodule is not installed locally

### Where are you running it?
- TFS Server 16.131.28601.4

### Version of Extension/Task
9.1.10

### Expected behaviour and actual behaviour
Pester task runs without throwing errors even when it is not able to check for updates or find Pester installed locally - in my understanding it should use the builtin executable. Actual behaviour: The tests are run, but errors are thrown, which causes the task to fail even if all tests were successsful.
Logs:
```powershell
2020-05-08T16:46:58.0927609Z ##[section]Starting: Pester Test Runner
2020-05-08T16:46:58.1110758Z ==============================================================================
2020-05-08T16:46:58.1110850Z Task         : Pester Test Runner
2020-05-08T16:46:58.1110964Z Description  : Run Pester tests by either installing the latest version of Pester at run time (if possible) or using the version shipped with the task (4.6.0)
2020-05-08T16:46:58.1111085Z Version      : 9.1.10
2020-05-08T16:46:58.1111125Z Author       : Pester
2020-05-08T16:46:58.1111173Z Help         : Version: #{Build.BuildNumber}#. [More Information](https://github.com/pester/AzureDevOpsExtension)
2020-05-08T16:46:58.1111242Z ==============================================================================
2020-05-08T16:46:58.3870412Z Using executable 'powershell.exe'
2020-05-08T16:46:58.3871527Z powershell.exe D:\development20181\_work\_tasks\Pester_cca5462b-887d-4617-bf3f-dcf0d3c622e9\9.1.10\Pester.ps1 -scriptFolder D:\development20181\_work\r1\a\XXX\Infrastructure\PsTools\RiStatus.Tests.ps1 -resultsFile D:\development20181\_work\r1\a\PesterRiTest.XML -run32Bit false
2020-05-08T16:46:59.2709271Z scriptFolder D:\development20181\_work\r1\a\XXX\Infrastructure\PsTools\RiStatus.Tests.ps1
2020-05-08T16:46:59.2710870Z resultsFile D:\development20181\_work\r1\a\PesterRiTest.XML
2020-05-08T16:46:59.2710954Z run32Bit false
2020-05-08T16:46:59.2711086Z additionalModulePath 
2020-05-08T16:46:59.2711211Z tag 
2020-05-08T16:46:59.2711306Z ExcludeTag 
2020-05-08T16:46:59.2712138Z CodeCoverageOutputFile 
2020-05-08T16:46:59.2712252Z CodeCoverageFolder 
2020-05-08T16:46:59.2712369Z ScriptBlock 
2020-05-08T16:47:02.1602366Z WARNING: MSG:UnableToDownload «https://go.microsoft.com/fwlink/?LinkID=627338&clcid=0x409» «»
2020-05-08T16:47:02.1608485Z WARNING: Unable to download the list of available providers. Check your internet connection.
2020-05-08T16:47:11.7519408Z WARNING: Unable to resolve package source 'https://www.powershellgallery.com/api/v2'.
2020-05-08T16:47:11.8843233Z ##[error]PackageManagement\Find-Package : No match was found for the specified search criteria and module name 'Pester'. Try 
2020-05-08T16:47:11.8853626Z ##[error]PackageManagement\Find-Package : No match was found for the specified search criteria and module name 'Pester'. Try 
2020-05-08T16:47:11.8872961Z ##[error]Get-PSRepository to see all available registered module repositories.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\2.0.1\PSModule.psm1:8845 char:3
+         PackageManagement\Find-Package @PSBoundParameters | Microsoft ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Microsoft.Power...ets.FindPackage:FindPackage) [Find-Package], Exceptio 
   n
    + FullyQualifiedErrorId : NoMatchFoundForCriteria,Microsoft.PowerShell.PackageManagement.Cmdlets.FindPackage
2020-05-08T16:47:11.8875102Z ##[error]Get-PSRepository to see all available registered module repositories.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\2.0.1\PSModule.psm1:8845 char:3
+         PackageManagement\Find-Package @PSBoundParameters | Microsoft ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Microsoft.Power...ets.FindPackage:FindPackage) [Find-Package], Exceptio 
   n
    + FullyQualifiedErrorId : NoMatchFoundForCriteria,Microsoft.PowerShell.PackageManagement.Cmdlets.FindPackage
2020-05-08T16:47:13.7019084Z Running in AMD64 PowerShell
2020-05-08T16:47:13.7028660Z Running Pester from the folder [D:\development20181\_work\r1\a\XXX\Infrastructure\PsTools\RiStatus.Tests.ps1] output sent to [D:\development20181\_work\r1\a\PesterRiTest.XML]
2020-05-08T16:47:13.9829537Z Pester v4.10.1

```

### Steps to reproduce the problem
Have it run without throwing errors.

Hello, we are getting errors as well during the package discovery phase. We just want to use the shipped version of Pester 4.10.1
Azure DevOps Server Version Dev17.M153.3
```powershell
##[section]Starting: Pester Test Runner
==============================================================================
Task         : Pester Test Runner
Description  : Run Pester tests by either installing the latest version of Pester at run time (if possible) or using the version shipped with the task (4.10.1)
Version      : 9.2.0
Author       : Pester
Help         : Version: #{Build.BuildNumber}#. [More Information](https://github.com/pester/AzureDevOpsExtension)
==============================================================================
Using executable 'powershell.exe'
powershell.exe E:\Agents\04\_work\_tasks\Pester_cca5462b-887d-4617-bf3f-dcf0d3c622e9\9.2.0\Pester.ps1 -scriptFolder @{Path='E:\Agents\04\_work\638\s/azuresql.template.tests.ps1'; Parameters=@{ResourceGroupName='***'; ServicePrincipalPassword='***'; SubscriptionId=***'; ApplicationId='***'; TenantId='***'; TemplatePath='E:\Agents\04\_work\638\s/'}} -resultsFile E:\Agents\04\_work\638\s/azuresql.template.tests.XML -run32Bit False
scriptFolder System.Collections.Hashtable

resultsFile E:\Agents\04\_work\638\s/azuresql.template.tests.XML

run32Bit False

additionalModulePath 

tag 

ExcludeTag 

CodeCoverageOutputFile 

CodeCoverageFolder 

ScriptBlock 

WARNING: MSG:UnableToDownload 
«https://go.microsoft.com/fwlink/?LinkID=627338&clcid=0x409» «»

WARNING: Unable to download the list of available providers. Check your 
internet connection.

WARNING: Unable to resolve package source 
'https://www.powershellgallery.com/api/v2'.


##[error]PackageManagement\Find-Package : No match was found for the specified search 
criteria and module name 'Pester'. Try Get-PSRepository to see all available 

##[error]PackageManagement\Find-Package : No match was found for the specified search 
criteria and module name 'Pester'. Try Get-PSRepository to see all available 

##[error]registered module repositories.
At C:\Program 
Files\WindowsPowerShell\Modules\PowerShellGet\2.2.3\PSModule.psm1:8873 char:9
+         PackageManagement\Find-Package @PSBoundParameters | Microsoft ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Microsoft.Power...ets.FindPacka 
   ge:FindPackage) [Find-Package], Exception
    + FullyQualifiedErrorId : NoMatchFoundForCriteria,Microsoft.PowerShell.Pac 
   kageManagement.Cmdlets.FindPackage
 
##[error]registered module repositories.
At C:\Program 
Files\WindowsPowerShell\Modules\PowerShellGet\2.2.3\PSModule.psm1:8873 char:9
+         PackageManagement\Find-Package @PSBoundParameters | Microsoft ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Microsoft.Power...ets.FindPacka 
   ge:FindPackage) [Find-Package], Exception
    + FullyQualifiedErrorId : NoMatchFoundForCriteria,Microsoft.PowerShell.Pac 
   kageManagement.Cmdlets.FindPackage
 

Running in AMD64 PowerShell

Running Pester using the script parameter [
Name                           Value                                           
----                           -----                                           
Path                           E:\Agents\04\_work\638\s/azuresql.template.te...
Parameters                     {ServicePrincipalPassword, ApplicationId, Res...

] output sent to [E:\Agents\04\_work\638\s/azuresql.template.tests.XML]

Pester v4.10.1
Executing all tests in 'E:\Agents\04\_work\638\s/azuresql.template.tests.ps1'

Executing script E:\Agents\04\_work\638\s/azuresql.template.tests.ps1

  Describing Template validation

    [+] template passes validation check
 2.68s

Tests completed in 5.3s

Tests Passed: 1, 
Failed: 0, 
Skipped: 0, 
Pending: 0, 
Inconclusive: 0 

Pester Script finished
##[section]Finishing: Pester Test Runner

```
As you can see, we have 1 test and it passes, but the task fails with errors
![image](https://user-images.githubusercontent.com/27974655/84290444-48a43d80-ab09-11ea-932a-372d2271e404.png)

--- a/Extension/PesterTask/PesterV9/HelperModule.psm1
+++ b/Extension/PesterTask/PesterV9/HelperModule.psm1
@@ -38,32 +38,32 @@ function Import-Pester {
         (Get-PSRepository)) {
 
         try {
-            $null = Get-PackageProvider -Name NuGet -ErrorAction Stop
-        }
-        catch {
             try {
-                Install-PackageProvider -Name Nuget -RequiredVersion 2.8.5.208 -Scope CurrentUser -Force -Confirm:$false -ErrorAction Stop
+                $null = Get-PackageProvider -Name NuGet -ErrorAction Stop
             }
             catch {
-                Write-Host "##vso[task.logissue type=warning]Falling back to version of Pester shipped with extension. To use a newer version please update the version of PowerShellGet available on this machine."
-                Import-Module -Name (Join-Path $PSScriptRoot "4.10.1\Pester.psd1") -Force -Verbose:$false -Global
+                Install-PackageProvider -Name Nuget -RequiredVersion 2.8.5.208 -Scope CurrentUser -Force -Confirm:$false -ErrorAction Stop
             }
-        }
 
-        if ($Version -eq "latest") {
-            $NewestPester = Find-Module -Name Pester -MaximumVersion 4.99.99 | Sort-Object Version -Descending | Select-Object -First 1
+            if ($Version -eq "latest") {
+                $NewestPester = Find-Module -Name Pester -MaximumVersion 4.99.99 -ErrorAction Stop | Sort-Object Version -Descending | Select-Object -First 1
 
-            If ((Get-Module Pester -ListAvailable | Sort-Object Version -Descending| Select-Object -First 1).Version -lt $NewestPester.Version) {
+                If ((Get-Module Pester -ListAvailable | Sort-Object Version -Descending| Select-Object -First 1).Version -lt $NewestPester.Version) {
+                    Install-Module -Name Pester -RequiredVersion $NewestPester.Version -Scope CurrentUser -Force -Repository $NewestPester.Repository -SkipPublisherCheck
+                }
+            }
+            else {
+                $NewestPester = Find-Module -Name Pester -RequiredVersion $Version -ErrorAction Stop | Select-Object -First 1
+
                 Install-Module -Name Pester -RequiredVersion $NewestPester.Version -Scope CurrentUser -Force -Repository $NewestPester.Repository -SkipPublisherCheck
             }
-        }
-        else {
-            $NewestPester = Find-Module -Name Pester -RequiredVersion $Version | Select-Object -First 1
 
-            Install-Module -Name Pester -RequiredVersion $NewestPester.Version -Scope CurrentUser -Force -Repository $NewestPester.Repository -SkipPublisherCheck
+            Import-Module -Name Pester -RequiredVersion $NewestPester.Version -Verbose:$false -Global
         }
-
-        Import-Module -Name Pester -RequiredVersion $NewestPester.Version -Verbose:$false -Global
+        catch {
+            Write-Host "##vso[task.logissue type=warning]Falling back to version of Pester shipped with extension. To use a newer version please update the version of PowerShellGet available on this machine."
+            Import-Module -Name (Join-Path $PSScriptRoot "4.10.1\Pester.psd1") -Force -Verbose:$false -Global
+        }
     }
     else {
         Write-Host "##vso[task.logissue type=warning]Falling back to version of Pester shipped with extension. To use a newer version please update the version of PowerShellGet available on this machine."

--- a/Extension/PesterTask/test/PesterTask/Helper.Tests.ps1
+++ b/Extension/PesterTask/test/PesterTask/Helper.Tests.ps1
@@ -2,7 +2,7 @@
 Describe "Testing Helper Functions" {
 
     BeforeAll {
-        Import-Module -Name (Resolve-Path "$PSScriptRoot/../../task/HelperModule.psm1") -Force
+        Import-Module -Name (Resolve-Path "$PSScriptRoot/../../PesterV9/HelperModule.psm1") -Forc
     }
     Context "Testing Get-HashtableFromString" {
 
@@ -82,103 +82,105 @@ Describe "Testing Helper Functions" {
         }
     }
 
-    Context "Testing Import-Pester" {
+    InModuleScope "HelperModule" {
+        Context "Testing Import-Pester" {
 
-        BeforeAll {
-            Mock -CommandName Import-Module -MockWith { }
-            Mock -CommandName Install-Module -MockWith { $true }
-            Mock -CommandName Write-host -MockWith { }
-            Mock -CommandName Get-PSRepository -MockWith {[PSCustomObject]@{Name = 'PSGallery'}}
-            Mock -CommandName Get-Command -MockWith { [PsCustomObject]@{Parameters=@{SkipPublisherCheck='SomeValue'}}} -ParameterFilter {$Name -eq 'Install-Module'}
-            Mock -CommandName Get-Command -MockWith { [PsCustomObject]@{Parameters=@{AllowPrerelease='SomeValue'}}} -ParameterFilter {$Name -eq 'Find-Module'}
-        }
-
-        It "Installs the latest version of Pester when on PS5+ and PowerShellGet is available" {
-            Mock -CommandName Find-Module -MockWith { [PsCustomObject]@{Version=[version]::new(9,9,9);Repository='PSGallery'}}
-            Mock -CommandName Get-PackageProvider -MockWith { $True }
-
-            Import-Pester -Version "latest"
-            
-            Assert-MockCalled -CommandName Import-Module -ParameterFilter {$RequiredVersion -eq "9.9.9"} -Scope It -Times 1
-        }
-
-        It "Installs the latest version of Pester from PSGallery when multiple repositories are available" {
-            Mock -CommandName Find-Module -MockWith { @(
-                    [PsCustomObject]@{Version=[version]::new(4,3,0);Repository='OtherRepository'}
-                    [PsCustomObject]@{Version=[version]::new(9,9,9);Repository='PSGallery'}
-                )
+            BeforeAll {
+                Mock -CommandName Import-Module -MockWith { }
+                Mock -CommandName Install-Module -MockWith { $true }
+                Mock -CommandName Write-host -MockWith { }
+                Mock -CommandName Get-PSRepository -MockWith {[PSCustomObject]@{Name = 'PSGallery'}}
+                Mock -CommandName Get-Command -MockWith { [PsCustomObject]@{Parameters=@{SkipPublisherCheck='SomeValue'}}} -ParameterFilter {$Name -eq 'Install-Module'}
+                Mock -CommandName Get-Command -MockWith { [PsCustomObject]@{Parameters=@{AllowPrerelease='SomeValue'}}} -ParameterFilter {$Name -eq 'Find-Module'}
             }
-            Mock -CommandName Get-PackageProvider -MockWith { $True }
 
-            Import-Pester -Version "latest"
+            It "Installs the latest version of Pester when on PS5+ and PowerShellGet is available" {
+                Mock -CommandName Find-Module -MockWith { [PsCustomObject]@{Version=[version]::new(9,9,9);Repository='PSGallery'}}
+                Mock -CommandName Get-PackageProvider -MockWith { $True }
 
-            Assert-MockCalled -CommandName Install-Module -Scope It -ParameterFilter {$Repository -eq 'PSGallery'}
-        }
-
-        It "Installs the required version of NuGet provider when PowerShellGet is available and NuGet isn't already installed" {
-            Mock -CommandName Find-Module -MockWith { [PsCustomObject]@{Version=[version]::new(9,9,9);Repository='PSGallery'}}
-            Mock -CommandName Get-PackageProvider -MockWith { throw }
-            Mock -CommandName Install-PackageProvider -MockWith {}
-
-            Import-Pester -Version "latest"
-
-            Assert-MockCalled -CommandName Install-PackageProvider
-        }
-
-        It "Should not install a new version of Pester when the latest is already installed" {
-            Mock -CommandName Find-Module -MockWith { [PsCustomObject]@{Version=(Get-Module Pester).Version;Repository='PSGallery'}}
-            Mock -CommandName Get-PackageProvider -MockWith { $True }
-
-            Import-Pester -Version "latest"
-
-            Assert-MockCalled -CommandName Install-Module -Times 0 -Scope It
-        }
-
-        It "Should install and import the specified version of Pester regardless of what is avaialble locally" {
-            Mock -CommandName Find-Module -MockWith { [PsCustomObject]@{Version=[version]::new(4,2,0);Repository='PSGallery'}}
-            Mock -CommandName Get-PackageProvider -MockWith { $True }
-
-            Import-Pester -Version 4.2.0
-
-            Assert-MockCalled -CommandName Install-Module -Times 1 -ParameterFilter { $RequiredVersion -eq "4.2.0"}
-            Assert-MockCalled -CommandName Import-Module -Times 1 -ParameterFilter {$RequiredVersion -eq "4.2.0"}
-        }
-
-        It "Should not Install the latest version of Pester when on PowerShellGet is available but SkipPublisherCheck is not available" {
-            Mock -CommandName Find-Module -MockWith { [PsCustomObject]@{Version=[version]::new(9,9,9);Repository='PSGallery'}}
-            Mock -CommandName Get-PackageProvider -MockWith { $True }
-            Mock -CommandName Get-Command -MockWith { [PsCustomObject]@{Parameters=@{OtherProperty='SomeValue'}} } -ParameterFilter {$Name -eq 'Install-Module'}
-
-            Import-Pester -Version "latest"
-
-            Assert-MockCalled -CommandName Install-Module -Times 0 -Scope It
-            Assert-MockCalled -CommandName Import-Module -Times 1 -ParameterFilter {$Name -like '*\4.10.1\Pester.psd1'}
-        }
-
-        It "Should fall back to build in version of Pester when no repositories are available" {
-            Mock -CommandName Get-PSRepository -MockWith {}
-            Mock -CommandName Get-Command -MockWith { [PsCustomObject]@{Parameters=@{SkipPublisherCheck='SomeValue'}}} -ParameterFilter {$Name -eq 'Install-Module'}
-
-            Import-Pester -Version "latest"
-
-            Assert-MockCalled -CommandName Install-Module -Times 0 -Scope It
-            Assert-MockCalled -CommandName Write-Host -Times 1 -Scope It -ParameterFilter {
-                $Object -eq "##vso[task.logissue type=warning]Falling back to version of Pester shipped with extension. To use a newer version please update the version of PowerShellGet available on this machine."
+                Import-Pester -Version "latest"
+                
+                Assert-MockCalled -CommandName Import-Module -ParameterFilter {$RequiredVersion -eq "9.9.9"} -Scope It -Times 1
             }
-            Assert-MockCalled -CommandName Import-Module -Times 1 -ParameterFilter {$Name -like '*\4.10.1\Pester.psd1'}
+
+            It "Installs the latest version of Pester from PSGallery when multiple repositories are available" {
+                Mock -CommandName Find-Module -MockWith { @(
+                        [PsCustomObject]@{Version=[version]::new(4,3,0);Repository='OtherRepository'}
+                        [PsCustomObject]@{Version=[version]::new(9,9,9);Repository='PSGallery'}
+                    )
+                }
+                Mock -CommandName Get-PackageProvider -MockWith { $True }
+
+                Import-Pester -Version "latest"
+
+                Assert-MockCalled -CommandName Install-Module -Scope It -ParameterFilter {$Repository -eq 'PSGallery'}
+            }
+
+            It "Installs the required version of NuGet provider when PowerShellGet is available and NuGet isn't already installed" {
+                Mock -CommandName Find-Module -MockWith { [PsCustomObject]@{Version=[version]::new(9,9,9);Repository='PSGallery'}}
+                Mock -CommandName Get-PackageProvider -MockWith { throw }
+                Mock -CommandName Install-PackageProvider -MockWith {}
+
+                Import-Pester -Version "latest"
+
+                Assert-MockCalled -CommandName Install-PackageProvider
+            }
+
+            It "Should not install a new version of Pester when the latest is already installed" {
+                Mock -CommandName Find-Module -MockWith { [PsCustomObject]@{Version=(Get-Module Pester).Version;Repository='PSGallery'}}
+                Mock -CommandName Get-PackageProvider -MockWith { $True }
+
+                Import-Pester -Version "latest"
+
+                Assert-MockCalled -CommandName Install-Module -Times 0 -Scope It
+            }
+
+            It "Should install and import the specified version of Pester regardless of what is avaialble locally" {
+                Mock -CommandName Find-Module -MockWith { [PsCustomObject]@{Version=[version]::new(4,2,0);Repository='PSGallery'}}
+                Mock -CommandName Get-PackageProvider -MockWith { $True }
+
+                Import-Pester -Version 4.2.0
+
+                Assert-MockCalled -CommandName Install-Module -Times 1 -ParameterFilter { $RequiredVersion -eq "4.2.0"}
+                Assert-MockCalled -CommandName Import-Module -Times 1 -ParameterFilter {$RequiredVersion -eq "4.2.0"}
+            }
+
+            It "Should not Install the latest version of Pester when on PowerShellGet is available but SkipPublisherCheck is not available" {
+                Mock -CommandName Find-Module -MockWith { [PsCustomObject]@{Version=[version]::new(9,9,9);Repository='PSGallery'}}
+                Mock -CommandName Get-PackageProvider -MockWith { $True }
+                Mock -CommandName Get-Command -MockWith { [PsCustomObject]@{Parameters=@{OtherProperty='SomeValue'}} } -ParameterFilter {$Name -eq 'Install-Module'}
+
+                Import-Pester -Version "latest"
+
+                Assert-MockCalled -CommandName Install-Module -Times 0 -Scope It
+                Assert-MockCalled -CommandName Import-Module -Times 1 -ParameterFilter {$Name -like '*\4.10.1\Pester.psd1'}
+            }
+
+            It "Should fall back to build in version of Pester when no repositories are available" {
+                Mock -CommandName Get-PSRepository -MockWith {}
+                Mock -CommandName Get-Command -MockWith { [PsCustomObject]@{Parameters=@{SkipPublisherCheck='SomeValue'}}} -ParameterFilter {$Name -eq 'Install-Module'}
+
+                Import-Pester -Version "latest"
+
+                Assert-MockCalled -CommandName Install-Module -Times 0 -Scope It
+                Assert-MockCalled -CommandName Write-Host -Times 1 -Scope It -ParameterFilter {
+                    $Object -eq "##vso[task.logissue type=warning]Falling back to version of Pester shipped with extension. To use a newer version please update the version of PowerShellGet available on this machine."
+                }
+                Assert-MockCalled -CommandName Import-Module -Times 1 -ParameterFilter {$Name -like '*\4.10.1\Pester.psd1'}
+            }
+
+            <#It "Loads Pester version that ships with task when not on PS5+ or PowerShellGet is unavailable" {
+                Mock -CommandName Invoke-Pester -MockWith { }
+                Mock -CommandName Import-Module -MockWith { }
+                Mock -CommandName Write-Host -MockWith { }
+                Mock -CommandName Write-Warning -MockWith { }
+                Mock -CommandName Write-Error -MockWith { }
+                Mock -CommandName Get-Module -MockWith { }
+
+                &$sut -ScriptFolder TestDrive:\ -ResultsFile TestDrive:\output.xml
+                Assert-MockCalled  Import-Module -ParameterFilter { $Name -eq "$pwd\4.10.1\Pester.psd1" }
+                Assert-MockCalled Invoke-Pester
+            }#>
         }
-
-        <#It "Loads Pester version that ships with task when not on PS5+ or PowerShellGet is unavailable" {
-            Mock -CommandName Invoke-Pester -MockWith { }
-            Mock -CommandName Import-Module -MockWith { }
-            Mock -CommandName Write-Host -MockWith { }
-            Mock -CommandName Write-Warning -MockWith { }
-            Mock -CommandName Write-Error -MockWith { }
-            Mock -CommandName Get-Module -MockWith { }
-
-            &$sut -ScriptFolder TestDrive:\ -ResultsFile TestDrive:\output.xml
-            Assert-MockCalled  Import-Module -ParameterFilter { $Name -eq "$pwd\4.10.1\Pester.psd1" }
-            Assert-MockCalled Invoke-Pester
-        }#>
-    }
+    } 
 }

--- a/Extension/PesterTask/test/PesterTask/Helper.Tests.ps1
+++ b/Extension/PesterTask/test/PesterTask/Helper.Tests.ps1
@@ -126,6 +126,21 @@ Describe "Testing Helper Functions" {
                 Assert-MockCalled -CommandName Install-PackageProvider
             }
 
+            It "Should fall back to build in version of Pester when Find-Module can't find Pester" {
+                Mock -CommandName Find-Module -MockWith { Write-Error "No match was found for the specified search 
+                criteria and module name 'Pester'"}
+                Mock -CommandName Get-PackageProvider -MockWith { $True }
+
+                Import-Pester -Version "latest"
+                
+                Assert-MockCalled -CommandName Install-Module -Times 0 -Scope It
+                Assert-MockCalled -CommandName Find-Module -Times 1 -Scope It
+                Assert-MockCalled -CommandName Write-Host -Times 1 -Scope It -ParameterFilter {
+                    $Object -eq "##vso[task.logissue type=warning]Falling back to version of Pester shipped with extension. To use a newer version please update the version of PowerShellGet available on this machine."
+                }
+                Assert-MockCalled -CommandName Import-Module -Times 1 -ParameterFilter {$Name -like '*\4.10.1\Pester.psd1'}
+            }
+
             It "Should not install a new version of Pester when the latest is already installed" {
                 Mock -CommandName Find-Module -MockWith { [PsCustomObject]@{Version=(Get-Module Pester).Version;Repository='PSGallery'}}
                 Mock -CommandName Get-PackageProvider -MockWith { $True }

--- a/Extension/PesterTask/test/PesterTask/Helper.Tests.ps1
+++ b/Extension/PesterTask/test/PesterTask/Helper.Tests.ps1
@@ -2,7 +2,7 @@
 Describe "Testing Helper Functions" {
 
     BeforeAll {
-        Import-Module -Name (Resolve-Path "$PSScriptRoot/../../PesterV9/HelperModule.psm1") -Forc
+        Import-Module -Name (Resolve-Path "$PSScriptRoot/../../PesterV9/HelperModule.psm1") -Force
     }
     Context "Testing Get-HashtableFromString" {
 

